.github/workflows/rust.yml

```diff
@@ -8,60 +8,47 @@ on:
 
 env:
   CARGO_INCREMENTAL: 0
+  Z3_RELEASE: 'z3-4.8.12'
+  RUST_BACKTRACE: 'full'
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - name: Install Z3
-      run: sudo apt-get install libz3-dev
-    - name: Build
-      run: cargo build -vv --all
-      # XXX: Ubuntu's Z3 package seems to be missing some symbols, like
-      # `Z3_mk_pbeq`, leading to linker errors. Just ignore this, I guess, until
-      # we figure out how to work around it. At least we have the
-      # statically-linked Z3 tests below...
-      if: ${{ success() || failure() }}
-    - name: Run tests
-      run: cargo test -vv --all
-      # See above.
-      if: ${{ success() || failure() }}
-    - name: Run tests with `arbitrary-size-numeral` enabled
-      run: cargo test --manifest-path z3/Cargo.toml -vv --features arbitrary-size-numeral
-      # See above.
-      if: ${{ success() || failure() }}
-
-  build_z3_statically:
     strategy:
       matrix:
-        build: [linux, macos, windows]
-        include:
-          - build: linux
-            os: ubuntu-latest
-          - build: macos
-            os: macos-latest
-          - build: windows
-            os: windows-latest
+        os: [ubuntu-latest, windows-latest]
```

Comment:
We would need macOS coverage here as well.

Reply:
I have never used macos in my life, do you know how one would install z3 system-wide there?

Related: I just noticed we don't have any guarantee that the preinstalled z3 on the linux machines have a compatible z3 version installed. Can/should we uninstall the preinstalled z3 and install the correct version for `dynamic-link-z3` on linux?